extensions/interactions/SetInput/directives/oppia-interactive-set-input.component.ts

```diff
@@ -0,0 +1,136 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the SetInput interaction.
+ *
+ * IMPORTANT NOTE: The naming convention for customization args that are passed
+ * into the directive is: the name of the parameter, followed by 'With',
+ * followed by the name of the arg.
+ */
+
+import { Component, Input, OnInit } from '@angular/core';
+import { InteractionAttributesExtractorService } from 'interactions/interaction-attributes-extractor.service';
+import { CurrentInteractionService } from 'pages/exploration-player-page/services/current-interaction.service';
+import { SetInputRulesService } from './set-input-rules.service';
+import eq from 'lodash/eq';
```

Comment:
Why do we need to import this? Can we implement this easily ourselves?

Reply:
I had to remove angular.equals and I didn't want to introduce more code that we had to maintain. (Plus the confidence on my PRs passing e2e tests has been low 😛, so I opted for this). I don't think this would affect our bundle size by much. Is there any other concern apart from the bundle size?

Reply:
Okay, seems that this should only import the code specific to `lodash/eq` so it should be fine.